contracts/cw-auth-middleware/src/contract.rs

```diff
@@ -0,0 +1,188 @@
+#[cfg(not(feature = "library"))]
+use cosmwasm_std::entry_point;
+use cosmwasm_std::{
+    to_binary, wasm_execute, Binary, CosmosMsg, Deps, DepsMut, Empty, Env, MessageInfo, Reply,
+    Response, StdError, StdResult, WasmMsg,
+};
+use cw2::set_contract_version;
+
+use crate::msg::IsAuthorizedResponse;
+use crate::state::Authorization;
+use crate::{
+    error::ContractError,
+    msg::{ExecuteMsg, InstantiateMsg, QueryMsg},
+    state::{Config, AUTHORIZATIONS, CONFIG},
+};
+
+const CONTRACT_NAME: &str = "crates.io:cw-auth-manager";
+const CONTRACT_VERSION: &str = env!("CARGO_PKG_VERSION");
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn instantiate(
+    deps: DepsMut,
+    _env: Env,
+    info: MessageInfo,
+    _msg: InstantiateMsg,
+) -> Result<Response, ContractError> {
+    set_contract_version(deps.storage, CONTRACT_NAME, CONTRACT_VERSION)?;
+    let config = Config {
+        dao: info.sender.clone(),
+    };
+    let empty: Vec<Authorization> = vec![];
+    CONFIG.save(deps.storage, &config)?;
+    AUTHORIZATIONS.save(deps.storage, &info.sender, &empty)?;
+
+    Ok(Response::default().add_attribute("action", "instantiate"))
+}
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn execute(
+    deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+    msg: ExecuteMsg,
+) -> Result<Response, ContractError> {
+    match msg {
+        ExecuteMsg::AddAuthorization { auth_contract } => {
+            execute_add_authorization(deps, env, info, auth_contract)
+        }
+        ExecuteMsg::Authorize { msgs, sender } => execute_authorize(deps, msgs, sender),
+    }
+}
+
+pub fn authorize_message(
+    contract: impl Into<String>,
+    msgs: &Vec<CosmosMsg>,
+    sender: impl Into<String>,
+) -> Result<WasmMsg, StdError> {
+    wasm_execute(
+        contract.into(),
+        &ExecuteMsg::Authorize {
+            msgs: msgs.clone(),
+            sender: sender.into(),
+        },
+        vec![],
+    )
+}
+
+fn execute_authorize(
+    deps: DepsMut,
+    msgs: Vec<CosmosMsg<Empty>>,
+    sender: String,
+) -> Result<Response, ContractError> {
+    let config = CONFIG.load(deps.storage)?;
+    let auths = AUTHORIZATIONS.load(deps.storage, &config.dao)?;
+
+    if auths.is_empty() {
```

Comment:
We could remove this. My thought was that we would want the auth contract to be there so that it's easy to configure it by adding and removing rules but that if it isn't configured it would just behave as if it wasn't there, but I understand how it can be confusing.

I actually just realized that removing this would actually leat to the same outcome, as no sub_msg would get added and the `Ok()` would be returned by default. So the solution would be to still check `is_empy()` but return an error. Just slightly different semantics. Happy to change it either way.